cmd/image-builder/main.go

```diff
@@ -156,6 +159,63 @@ func cmdManifest(cmd *cobra.Command, args []string) error {
 	return err
 }
 
+var awscloudNewUploader = awscloud.NewUploader
+
+func cmdUploadAWS(cmd *cobra.Command, args []string) error {
+	amiName, err := cmd.Flags().GetString("aws-ami-name")
+	if err != nil {
+		return err
+	}
+	bucketName, err := cmd.Flags().GetString("aws-bucket")
+	if err != nil {
+		return err
+	}
+	region, err := cmd.Flags().GetString("aws-region")
+	if err != nil {
+		return err
+	}
+
+	rawDiskPath := args[0]
+	// XXX: can we actually inspect the image or leave some artifacts?
+	if filepath.Ext(rawDiskPath) != ".raw" {
+		return fmt.Errorf("expecting a raw disk ending with '.raw', got %q", filepath.Base(rawDiskPath))
+	}
```

Comment:
Not sure about this.  Yeah, `.raw` is conventional but do we really want to enforce it?  Especially since we have plans to let people set the name for the artifact, and I've seen cases where people will choose `.img` or similar.  It'd be much nicer to look for the PT header if we want to do it right.  Either that or not check it at all.

Reply:
I now realise that this requirement didn't make it into the final patch, but the extension is still used to identify the provider to upload to, so I see the same issue there.

Reply:
Thank you, that makes sense, I improved the detection based on the actual content, I hope this is what you had in mind. We could also make this explicit and ask the user to provide a type when uploading but that would add another parameter. I can also squash the last commit into the previous one(s) if you want.

Reply:
For upload this is now an explicit user choice what cloud to use, this should also address the concerns from ondrej that people may upload non-raw images.

Reply:
> Especially since we have plans to let people set the name for the artifact, and I've seen cases where people will choose `.img` or similar.

I'm not sure if it is a good idea to let people specify the extension. My expectation would be that customizing the filename does not affect the extension :innocent:

I tend to agree that checking the extension does not seem to add much if one needs to explicitly specify the cloud to upload to.